### PR TITLE
adds filter fields to cache key name hash

### PR DESCRIPTION
**Pull request by justin-cechmanek**
This PR aims to prevent cache entries being overwritten when the exact same prompt string is used, but other cache entry data - like user id, or some other field - is different.

### CODE REVIEW

File: redisvl/redis/utils.py
Comment (tylerhutcherson) [inline redisvl/redis/utils.py:45]: Small preference here to keep `hashify` dumb and move the filter string processing logic for this to within the `generate_id` method and pass the final string to hashify.

**Review by tylerhutcherson — APPROVED** [review]
LGTM! one small note, but good to go when you're ready. Once merged, can you close out that issue and follow up internally too?

**Review by rbs333 — APPROVED** [review]

**Pull request merged**
